R/explorer.R

```diff
@@ -26,45 +49,58 @@ explorer <- function(dataList=c(), addEnv=T, demo=F) {
     )
   )
 
-  formatFileList<-function(fileList){
-    fileList_formatted = list()
-    i<-1
-    for(f in fileList){
-      df<-get(f)
-      if(class(df)=="data.frame"){
-        fileObj = list(
-          File=f,
-          Rows = dim(df)[1],
-          Columns = dim(df)[2],
+  formatFileList <- function(fileList){
+    fileList_formatted <- list()
+    if (is.list(fileList)){
+      for (i in seq_along(fileList)){
+        df <- fileList[[i]]
+        fileList_formatted[[i]] <- list(
+          File = namesList[i],
+          Rows = nrow(df),
+          Columns = ncol(df),
+          json = jsonlite::toJSON(df)
+        )
+      }      
+    } else if (is.vector(fileList)){
+      for (i in seq_along(fileList)){
+        df <- get(fileList[i])
+        fileList_formatted[[i]] <- list(
+          File = fileList[i],
+          Rows = nrow(df),
+          Columns = ncol(df),
           json = jsonlite::toJSON(df)
         )
-        fileList_formatted[[i]]<-fileObj
-        i<-i+1
       }
     }
     return(fileList_formatted)
   }
 
+  
   # (2) Prep an array of objects for the user-specified files  (if length(dataArray)>0)
     data_list_formatted <- formatFileList(dataList)
```

Comment:
I thought about doing that, but we still need a vector-specific function for addEnv = TRUE and demo=TRUE.  Those are character vectors and I didn't really want to bring them in as a list of dataframes, because we end up taking up more memory than needed...